modules/nickserv/cert.c

```diff
@@ -167,14 +175,23 @@ ns_cmd_cert(struct sourceinfo *si, int parc, char *parv[])
 			command_fail(si, fault_noprivs, STR_NOT_LOGGED_IN);
 			return;
 		}
-		cert = mycertfp_find(parv[1]);
+
+		struct hook_user_certfp hdata = {
+			.si = si,
+			.mu = mu
+		};
+		mowgli_strlcpy(hdata.certfp, parv[1], sizeof hdata.certfp);
+		hook_call_user_certfp_del(&hdata);
+
+		cert = mycertfp_find(hdata.certfp);
```

Comment:
If the `user_certfp_del` hook changed the certfp, we probably want to check for the unmodified certfp first, falling back to the modified certfp; otherwise, users may be unable to easily remove a fingerprint from their list if it is in a format the hook will modify.

(This issue can come up if the add hook doesn't match the delete hook, or more typically, if the user added a fingerprint at a time when the delete hook was not present.)